core/trino-spi/src/main/java/io/trino/spi/security/SystemAccessControl.java

```diff
@@ -942,6 +943,22 @@ default Optional<ViewExpression> getColumnMask(SystemSecurityContext context, Ca
         return Optional.empty();
     }
 
+    /**
+     * Bulk method for getting column masks for a subset of columns in a table.
+     * <p>
+     * Constraints from {@link SystemAccessControl#getColumnMask(SystemSecurityContext, CatalogSchemaTableName, String, Type)}
+     * carry over to each mask returned.
+     *
+     * @return a mapping from columns to masks. The keys of the return Map are a subset of {@code columns}.
+     */
+    default Map<ColumnSchema, ViewExpression> getTableColumnMasks(SystemSecurityContext context, CatalogSchemaTableName tableName, List<ColumnSchema> columns)
+    {
+        return columns.stream()
+                .map(c -> Map.entry(c, getColumnMask(context, tableName, c.getName(), c.getType())))
+                .filter(entry -> entry.getValue().isPresent())
+                .collect(Collectors.toMap(Map.Entry::getKey, entry -> entry.getValue().get()));
```

Comment:
I'm not sure if this should have a default that calls `getColumnMask`, looking at other function in this interface none of them do this. Should this logic be left to another class?

Reply:
yes, the default should be to call the old method
the engine should call the new method only
the old method should be deprecated

Reply:
thanks! do we want to remove `getColumnMask` from the `AccessControl` interface? that would be the best way to ensure the engine is not using it anymore

Reply:
yes, remove the supposed-to-be-unused method from the interface defined in trino-main

deprecation is needed only for trino-spi's interfaces